platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/network/okhttp/CaptureOkHttpEventListener.kt

```diff
@@ -91,6 +91,10 @@ internal class CaptureOkHttpEventListener internal constructor(
     private var lastResponse: Response? = null
 
     override fun callStart(call: Call) {
+        if (call.shouldIgnoreCurrentPath()) {
+            return
```

Comment:
IIUC this check is all that would be really needed? since when true it would never initialize requestInfo and those null checks would take care of the rest no?

my suggestion would be to just have this one check (probably move it after the targetEventListener forwarding) and then greatly simplify this change